destination/destination.go

```diff
@@ -0,0 +1,125 @@
+// Copyright © 2023 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destination
+
+//go:generate paramgen -output=paramgen_dest.go Config
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/conduitio-labs/conduit-connector-grpc-client/destination/toproto"
+	pb "github.com/conduitio-labs/conduit-connector-grpc-client/proto/v1"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"google.golang.org/grpc"
+)
+
+type Destination struct {
+	sdk.UnimplementedDestination
+
+	config Config
+	conn   *grpc.ClientConn
+	stream pb.StreamService_StreamClient
+	client pb.StreamServiceClient
```

Comment:
Nitpick: Technically we don't need this field, we only use the client in `Open`, and it should probably stay like that.

Reply:
I'm gonna need it when I get to backoff retry, cuz I need to create a new stream each time it reconnects

Reply:
That depends on how we do the reconnecting. If we do it in `Write`, then yes. But IMO it would be better to have a lock that guards the `stream` field and spawn a goroutine in `Open` that continuously makes sure the connection is healthy. If the connection drops, it locks the stream and tries to reconnect and create a new stream.

This reminds me - if we want to make sure we notice a dropped connection even when no messages are flowing through the stream, we might need to include a heartbeat RPC, so we know that the other side is still alive. That goroutine I mentioned that we'd spawn in `Open` would keep sending and receiving heartbeats, and if the heartbeats start failing we know there's an issue with the connection.

Reply:
We can kick this down the line into a separate PR ☝️

Reply:
After a short research I'm not sure if we need heartbeats, we might get away with checking [`grpc.ClientConn.GetState`](https://pkg.go.dev/google.golang.org/grpc#ClientConn.GetState) and [`grpc.ClientConn.WaitForStateChange`](https://pkg.go.dev/google.golang.org/grpc#ClientConn.WaitForStateChange). Also there seems to be a [backoff](https://pkg.go.dev/google.golang.org/grpc@v1.54.0/backoff) mechanism built in that we can use for reconnecting. We need to do some testing to see if that already provides the behavior we want.

Reply:
yeah I used these two methods with my first approach, but had some problems with the timeout, I think now I know why I had those and can use the idea of spawning a go routine in Open and lock the stream to solve them! So I'll go with that.
oh and backoff is already implemented by default with the gRPC connection, we only need to change the default values if needed.